addons/AutoExportVersion/AutoExportVersion.gd

```diff
@@ -1,147 +1,185 @@
+
+## Version from an export profile                                                               [br]
+## The version will be the first non-empty version value from the first profile with that value.[br]
+## Useful for versions like '1.0.0'                                                             [br]
+## !!! Requires export_presets.cfg to exist.
+func get_export_preset_version() -> String:
+	const version_keys: Array[String] = [
+		"file_version", # Windows
+		"product_version", # Windows
+		"version/name", # Android
+		"version/code", # Android
+		"application/short_version", # Mac/iOS
+		"application/version", # Mac/iOS
+	]
 	
-	# Similar to profile version, but uses only Android version code and name.
-	# Edit it if you want to format the version differently.
+	var config: ConfigFile = ConfigFile.new()
+	var err: int = config.load("res://export_presets.cfg")
+	if err != OK:
+		push_error("Cannot open 'res://export_presets.cfg'. Error: %s" % error_string(err))
+		return ""
 	
-#	var config := ConfigFile.new()
-#	if config.load("res://export_presets.cfg") == OK:
-#		var code := ""
-#		var vname := ""
-#		var found: int
-#
-#		for section in config.get_sections():
-#			if section.ends_with(".options"):
-#				for key in config.get_section_keys(section):
-#					if key == "version/code":
-#						code = str(config.get_value(section, key))
-#						found |= 1
-#					elif key == "version/name":
-#						vname = str(config.get_value(section, key))
-#						found |= 2
-#
-#						if found == 3:
-#							break
-#				if found == 3:
-#					break
-#			if found == 3:
-#				break
-#
-#		if found != 3:
-#			push_error("Failed to fetch version. No valid version code and name found in export profiles.")
-#		else:
-#			# Edit formatting here.
-#			return "%s %s" % [code, vname]
-#	else:
-#		push_error("Failed to fetch version. export_presets.cfg does not exist.")
+	for section in config.get_sections():
+		if not section.ends_with(".options"):
+			continue
+		for key in config.get_section_keys(section):
+			for check_key in version_keys:
+				if key.ends_with(check_key):
+					var version: String =  str(config.get_value(section, key))
+					if version.is_empty():
+						continue 
+					return version
 	
+	push_error("Failed to fetch version. No valid version key found in export profiles.")
 	return ""
 
-### Unimportant stuff here.
+## Stores a [param version] based on [param version_store_location].                            [br]
+## See [member PROJECT_SETTING_NAME], [member SCRIPT_PATH]
+func store_version(version: String, version_store_location: VersionStoreLocation=VersionStoreLocation.PROJECT_SETTING) -> void:
+	if version.is_empty():
+		printerr("Cannot store version. " + _EMPTY_VERSION_ERROR.format({"script_path": get_script().get_path()}))
+		return
+	
+	match version_store_location:
+		VersionStoreLocation.SCRIPT:
+			store_version_as_script(version)
+		VersionStoreLocation.PROJECT_SETTING:
+			store_version_as_project_setting(version)
```

Comment:
Not consistent with enum order.

Reply:
Rearranged the related code to be in order in [this commit](https://github.com/Kubulambula/Godot-Auto-Export-Version/commit/fd5a12d0e6e94a80f14c742e8c90783d7d9e51df)